pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,350 @@
+		return err
+	}
+
+	// Remove run condition form the service
+	conditionOpt := &unit.UnitOption{
+		Section: "Unit",
+		Name:    "ConditionPathExists",
+		Value:   "!/etc/ignition-machine-config-encapsulated.json",
+	}
+	switchdevService, err = service.RemoveFromService(switchdevService, conditionOpt)
+	if err != nil {
+		return err
+	}
+	p.switchdevService = switchdevService
+
+	switchdevRunScript, err := service.ReadScriptManifestFile(configuresSwitchdevScript)
+	if err != nil {
+		return err
+	}
+
+	p.switchdevRunScript = switchdevRunScript
+
+	return nil
+}
+
+func (p *K8sPlugin) readNetworkManagerManifest() error {
+	networkManagerService, err := service.ReadServiceInjectionManifestFile(networkManagerUnitFile)
+	if err != nil {
+		return err
+	}
+
+	p.networkManagerService = networkManagerService
+	return nil
+}
+
+func (p *K8sPlugin) readOpenVSwichdManifest() error {
+	openVSwitchService, err := service.ReadServiceInjectionManifestFile(ovsUnitFile)
+	if err != nil {
+		return err
+	}
+
+	p.openVSwitchService = openVSwitchService
+	return nil
+}
+
+func (p *K8sPlugin) readManifestFiles() error {
+	if err := p.readSwitchdevManifest(); err != nil {
+		return err
+	}
+
+	if err := p.readNetworkManagerManifest(); err != nil {
+		return err
+	}
+
+	if err := p.readOpenVSwichdManifest(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) switchdevServiceStateUpdate() error {
+	// Check switchdev service
+	swdService, err := p.serviceManager.ReadService(p.switchdevService.Path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		// service not exists
+		p.updateTarget.switchdevService = true
+	} else {
+		needChange, err := service.CompareServices(swdService, p.switchdevService)
+		if err != nil {
+			return err
+		}
+		p.updateTarget.switchdevService = needChange
+	}
+
+	// Check switchdev run script
+	data, err := ioutil.ReadFile(path.Join(chroot, p.switchdevRunScript.Path))
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		p.updateTarget.switchdevScript = true
+	} else if string(data) != p.switchdevRunScript.Contents.Inline {
+		p.updateTarget.switchdevScript = true
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) getSystemServices() []*service.Service {
+	return []*service.Service{p.networkManagerService, p.openVSwitchService}
+}
+
+func (p *K8sPlugin) isSystemServiceNeedUpdate(serviceObj *service.Service) bool {
+	systemService, err := p.serviceManager.ReadService(serviceObj.Path)
+	if err != nil {
+		return false
+	}
+	if systemService != nil {
+		needChange, err := service.CompareServices(systemService, serviceObj)
+		if err != nil {
+			return false
+		}
+		return needChange
+	}
+
+	return false
+}
+
+func (p *K8sPlugin) systemServicesStateUpdate() error {
+	var services []*service.Service
+	for _, systemService := range p.getSystemServices() {
+		exist, err := p.serviceManager.IsServiceExist(systemService.Path)
+		if err != nil {
+			return err
+		}
+		if !exist {
+			continue
```

Comment:
if it not exist it a problem we should log error

Reply:
if it not exist it means the NetworkManager or ovs is not installed on the host so a warning would serve better unless you consider them a pre-request, WDYT?

Reply:
The ovs is must not sure about the NM

Reply:
So it would be better to split the logic to check if the service needs an update because there are mandatory and optional services

Reply:
Let assume all of are mandatory ovs and NM